game/src/main/scala/io/aigar/game/Player.scala

```diff
@@ -125,13 +123,15 @@ class Player(val id: Int, startPosition: Vector2) extends EntityContainer
    */
   def onExternalAction: Unit = {
     if (!isActive) logger.info(s"Player $id reconnected.")
-    aiState.onPlayerActivity
+    cells.foreach { _.aiState.onPlayerActivity }
   }
 
   /**
     * The player is active when there is an active cell.
     */
   def isActive(): Boolean = {
-    aiState.isActive
+    cells.map{ _.aiState.isActive }.fold(false) { (z, i) =>
```

Comment:
Both of your suggestions seem good. Does the ```cells.exists(_.aiState.isActive)``` will return false if `cells` is empty? And will `exists` return false if `isActive` exists but returns `false`?